internal/builtinfunctions/functions.go

```diff
@@ -164,7 +166,46 @@ func getFloatToStringFunction() schema.CallableFunction {
 	return funcSchema
 }
 
-// TODO: Webb can add a function called floatToStringAdvanced that allows other float formats
+func getFloatToFormattedStringFunction() schema.CallableFunction {
+	funcSchema, err := schema.NewCallableFunction(
+		"floatToFormattedString",
+		[]schema.Type{
+			schema.NewFloatSchema(nil, nil, nil),
+			schema.NewStringSchema(nil, nil, regexp.MustCompile(`^[beEfFgGxX]$`)),
+			schema.NewIntSchema(nil, nil, nil),
+		},
+		// 'b' format: -ddddp±ddd
+		// 'e' format: -d.dddde±dd
+		// 'E' format: -d.ddddE±dd
+		// 'f' format: -ddd.dddd
+		// 'x' format: -0xd.ddddp±ddd
+		// 'X' format: -0Xd.ddddP±ddd
+		schema.NewStringSchema(
+			nil,
+			nil,
+			regexp.MustCompile(`^-?(?:0[xX])?\d+(?:\.\d*)?(?:[pPeE][-+]/d{2,3})?$`)),
```

Comment:
Is the `/d{2,3}` supposed to be `\d{2,3}`, or do you really mean a literal `/` followed by 2 or 3 `d` characters? 😁